pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -1560,7 +1560,20 @@ async def run(self):
                     model_type = type(d_content.content)
                     unit_q = d_content.content.q()
                     content_q_by_type[model_type] = content_q_by_type[model_type] | unit_q
-                    d_content_by_nat_key[d_content.content.natural_key()].append(d_content)
+                    # ============ The below lines are added vs. pulpcore ============
```

Comment:
Why not just initialize the model entry in the stage to match the normalized form coming out of the database>

Reply:
Indeed. Got the whole issue when you sent the first line.